### PR TITLE
AST - Fix Earthly Star counts

### PR DESCRIPTION
**Pull request by CasualSuperman**
We weren't ever incrementing uses, which was the main issue.
Also copy the logic from Draw for calculating lost time if it stopped being used at some point during the fight.
Fixes issue reported on discord.

### CODE REVIEW

File: src/parser/jobs/ast/modules/Sect/Sect.tsx
Comment (nonowazu) [inline src/parser/jobs/ast/modules/Sect/Sect.tsx:146]: blank if section? Is there supposed to be logic here?

**Review by nonowazu — CHANGES_REQUESTED** [review]

File: src/parser/jobs/ast/index.tsx
Comment (CasualSuperman) [reply src/parser/jobs/ast/index.tsx:53]: It's meant to be separate - I made this (and most of the other ones) as a draft and was going to publish it after #1023 got merged so I wasn't adding myself as a contributor in multiple PRs, getting conflicts later, etc.

File: src/parser/jobs/ast/modules/EarthlyStar.tsx
Comment (CasualSuperman) [reply src/parser/jobs/ast/modules/EarthlyStar.tsx:32]: Whoops.

File: src/parser/jobs/ast/modules/Sect/Sect.tsx
Comment (CasualSuperman) [reply src/parser/jobs/ast/modules/Sect/Sect.tsx:146]: It was there just to document why the check was necessary.

File: src/parser/jobs/ast/index.tsx
Comment (nonowazu) [reply src/parser/jobs/ast/index.tsx:53]: That's perfectly fine, but it's confusing from a 'hey, which PR made this change' perspective when there's two changes in the history list under a job associated only with a singular PR.

File: src/parser/jobs/ast/modules/Sect/Sect.tsx
Comment (nonowazu) [reply src/parser/jobs/ast/modules/Sect/Sect.tsx:146]: I still find the blank logic confusing - code shouldn't be a substitute for comments, especially when it leads to noop-style logic to have 'self documenting code'. Instead of using a blank `if` block, document as part of a comment and invert the logic to be:

```js
if (event.timestamp >= this.parser.eventtimeOffset) {
```

File: src/parser/jobs/ast/index.tsx
Comment (AzariahAeroborn) [reply src/parser/jobs/ast/index.tsx:53]: Once you merge one, if @CasualSuperman back-merges master into the other branch, it will automatically remove the "duplicated" changeset from the other, in either order.

Comment (CasualSuperman) [reply src/parser/jobs/ast/index.tsx:53]: Agreed, I wasn't sure what the best route was (didn't even know drafts were publicly listed tbh)

File: src/parser/jobs/ast/modules/Sect/Sect.tsx
Comment (CasualSuperman) [reply src/parser/jobs/ast/modules/Sect/Sect.tsx:146]: Fixed this in #1023, since that's the PR those changes came from.

File: src/parser/jobs/ast/index.tsx
Comment (CasualSuperman) [reply src/parser/jobs/ast/index.tsx:53]: Would it help if I abandoned the dependent PRs, and reopened them once the prior changes have been landed?

File: src/parser/jobs/ast/modules/Sect/Sect.tsx
Comment (nonowazu) [reply src/parser/jobs/ast/modules/Sect/Sect.tsx:146]: I'll let this go, but in the future document when PRs have ordered dependencies, and please try not to base your PR off another PR branch - small atomic commits are much easier to reason about.

Comment (CasualSuperman) [reply src/parser/jobs/ast/modules/Sect/Sect.tsx:146]: I did say in the PR description that it was branched off of #1023..

Comment (nonowazu) [reply src/parser/jobs/ast/modules/Sect/Sect.tsx:146]: I understand this was branched off of #1023, but I didn't know about the ordered dependency until we discussed it., and the suggestion was to base your PRs off of master moving forward (unless there's a compelling reason not to).

Comment (CasualSuperman) [reply src/parser/jobs/ast/modules/Sect/Sect.tsx:146]: Shouldn't be an issue since this was based off of that to avoid duplicating/conflicting adding myself as a contributor, and I didn't add it to that one so I could keep the individual PR small. This was a bit of a one-off situation for me.

**Review by nonowazu — DISMISSED** [review]
I'm going to approve this because it appears that my requested changes were implemented, but Ay is going to give this another set of eyes on this because I usually only check for code sanity, not full logic flow.

**Comment by CasualSuperman** [comment]
I don't know who Ay is short for, can you tag them to see when they can give this a look? Someone joined the discord to ask about this bug.

**Comment by karashiiro** [comment]
@ayyaruq

File: src/parser/jobs/ast/modules/EarthlyStar.tsx
Comment (ayyaruq) [inline src/parser/jobs/ast/modules/EarthlyStar.tsx:81]: This is generally messy and could be done more simply with a clamp. In the interest of getting it through, ignoring, but at some point drift and also held times should probably be rethought. Could just initialise lastUse to be fight start time.

File: src/parser/jobs/ast/modules/EarthlyStar.tsx
Comment (ayyaruq) [inline src/parser/jobs/ast/modules/EarthlyStar.tsx:118]: Is it not accurate already? The held bump could prob be a method of it's own so you end up with less funky math like this. Again, won't block on this, but the whole module could use a cleanup in this regard.

**Review by ayyaruq — APPROVED** [review]
I'm taking some issue with code structure and dead code but it works so whatever. Could prob rethink the way this module does stuff as a whole at some point.

File: src/parser/jobs/ast/modules/EarthlyStar.tsx
Comment (CasualSuperman) [reply src/parser/jobs/ast/modules/EarthlyStar.tsx:81]: Easy enough to change

File: src/parser/jobs/ast/modules/EarthlyStar.tsx
Comment (CasualSuperman) [reply src/parser/jobs/ast/modules/EarthlyStar.tsx:118]: I think some form of logic for this is required.
Say they use Earthly Star once at the beginning of the fight, and never again. `onPlace` won't be called again, so nothing will increase the `totalHeld` variable, so at the end of the fight we'll need to see how many more they could have cast.

File: src/parser/jobs/ast/modules/EarthlyStar.tsx
Comment (CasualSuperman) [reply src/parser/jobs/ast/modules/EarthlyStar.tsx:32]: Nope, left it in from while I was figuring out what was going wrong. I'll remove it.

File: src/parser/jobs/ast/modules/EarthlyStar.tsx
Comment (ayyaruq) [reply src/parser/jobs/ast/modules/EarthlyStar.tsx:114]: So?

**Review by ayyaruq — APPROVED** [review]

**Comment by ayyaruq** [comment]
Merged so we can get this done without further superfluous edits.

File: src/parser/jobs/ast/modules/EarthlyStar.tsx
Comment (CasualSuperman) [reply src/parser/jobs/ast/modules/EarthlyStar.tsx:114]: Maybe I'm missing your point. It sounded like you thought this code wasn't required based on "Is it not accurate already?"

Comment (ayyaruq) [reply src/parser/jobs/ast/modules/EarthlyStar.tsx:114]: I get what it does, my point was about fixing the initial drift conditional and collapsing the 3 diff calculations into a single method that can be called in onPlace and onComplete to avoid duplicating complex-looking math and making it easier to maintain.

**Pull request merged**
